src/battle_controllers.c

```diff
@@ -0,0 +1,1551 @@
+
+    for (i = 0; i < size; i++)
+        gLinkBattleSendBuffer[gTasks[gUnknown_02022D08].data[14] + LINK_BUFF_DATA + i] = data[i];
+
+    gTasks[gUnknown_02022D08].data[14] = gTasks[gUnknown_02022D08].data[14] + alignedSize + LINK_BUFF_DATA;
+}
+
+static void Task_HandleSendLinkBuffersData(u8 taskId)
+{
+    u16 var;
+    u16 blockSize;
+
+    switch (gTasks[taskId].data[11])
+    {
+    case 0:
+        gTasks[taskId].data[10] = 100;
+        gTasks[taskId].data[11]++;
+        break;
+    case 1:
+        gTasks[taskId].data[10]--;
+        if (gTasks[taskId].data[10] == 0)
+            gTasks[taskId].data[11]++;
+        break;
+    case 2:
+        if (gLinkVSyncDisabled)
+        {
+            gTasks[taskId].data[11]++;
+        }
+        else
+        {
+            if (gBattleTypeFlags & BATTLE_TYPE_BATTLE_TOWER)
+                var = 2;
+            else
+                var = (gBattleTypeFlags & BATTLE_TYPE_MULTI) ? 4 : 2;
+
+            if (sub_800ABAC() >= var)
+            {
+                if (sub_800ABBC())
+                {
+                    sub_800A620();
+                    gTasks[taskId].data[11]++;
+                }
+                else
+                {
+                    gTasks[taskId].data[11]++;
+                }
+            }
+        }
+        break;
+    case 3:
+        if (gTasks[taskId].data[15] != gTasks[taskId].data[14])
+        {
+            if (gTasks[taskId].data[13] == 0)
+            {
+                if (gTasks[taskId].data[15] > gTasks[taskId].data[14]
+                 && gTasks[taskId].data[15] == gTasks[taskId].data[12])
+                {
+                    gTasks[taskId].data[12] = 0;
+                    gTasks[taskId].data[15] = 0;
+                }
+                blockSize = (gLinkBattleSendBuffer[gTasks[taskId].data[15] + LINK_BUFF_SIZE_LO] | (gLinkBattleSendBuffer[gTasks[taskId].data[15] + LINK_BUFF_SIZE_HI] << 8)) + LINK_BUFF_DATA;
+                SendBlock(bitmask_all_link_players_but_self(), &gLinkBattleSendBuffer[gTasks[taskId].data[15]], blockSize);
+                gTasks[taskId].data[11]++;
+            }
+            else
+            {
+                gTasks[taskId].data[13]--;
+                break;
+            }
+        }
+        break;
+    case 4:
+        if (sub_800A520())
+        {
+            blockSize = gLinkBattleSendBuffer[gTasks[taskId].data[15] + LINK_BUFF_SIZE_LO] | (gLinkBattleSendBuffer[gTasks[taskId].data[15] + LINK_BUFF_SIZE_HI] << 8);
+            gTasks[taskId].data[13] = 1;
+            gTasks[taskId].data[15] = gTasks[taskId].data[15] + blockSize + LINK_BUFF_DATA;
+            gTasks[taskId].data[11] = 3;
+        }
+        break;
+    case 5:
+        gTasks[taskId].data[13]--;
+        if (gTasks[taskId].data[13] == 0)
+        {
+            gTasks[taskId].data[13] = 1;
+            gTasks[taskId].data[11] = 3;
+        }
+        break;
+    }
+}
+
+// fix me
```

Comment:
Did you forget to fix this?

Reply:
Nope, I was unable to just as as the person who decompiled this function in Ruby. It's a fake match and the comment is there, so maybe someone fixes it in the future.